plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go

```diff
@@ -0,0 +1,498 @@
+		return ""
+	}
+
+	return sseURL + "/" + pl.Properties.ID
+}
+
+// reconnect is called when the connection is broken
+// reconnect starts restablishing the connection
+func (s *ctrlXDataLayerSSE) reconnect(sub subscription) {
+	go func() {
+		s.Log.Infof("Subscription %v is down, reconnect is starting", sub.index)
+		for {
+			time.Sleep(time.Duration(defaultReconnectInterval))
+
+			if !s.setAuthToken() {
+				continue
+			}
+
+			subURL := s.createSubscription(&sub)
+			if subURL == "" {
+				continue
+			}
+
+			client := newClient(subURL, sub, s.Log)
+			if s.autoReconnect {
+				client.onDisconnect(s.reconnect)
+			}
+			client.Headers["Authorization"] = s.token.String()
+			client.Connection.Transport = httpTransport(s.InsecureSkipVerify)
+			err := client.requestData(s, sub.Measurement)
+			if err != nil {
+				continue
+			}
+
+			s.Log.Infof("Subscription %v is reconnected", sub.index)
+			return
+		}
+	}()
+}
+
+// addMetric writes sse metric into accumulator
+func (s *ctrlXDataLayerSSE) addMetric(se *sSEEvent, measurement string) {
+	m := metric{}
+	switch se.Event {
+	case "update":
+		// Received an updated value, that we translate into a metric
+		_ = json.Unmarshal([]byte(se.Data), &m)
+		t := convertTimestamp2UnixTime(m.Timestamp)
+		if s.acc != nil {
+			s.wg.Add(1)
+			s.acc.AddFields(measurement, map[string]interface{}{"value": m.Value}, map[string]string{"node": m.Node, "type": m.Type, "host": s.Hostname}, t)
+			s.wg.Done()
+		}
+
+	case "error":
+		// Received an error event, that we report to the accumulator
+		em := errorMetric{}
+		_ = json.Unmarshal([]byte(se.Data), &em)
+		if s.acc != nil {
+			s.wg.Add(1)
+			s.acc.AddError(fmt.Errorf("Received error event for: %v", em.Instance))
+			s.wg.Done()
+		}
+
+	case "keepalive":
+		// Keepalive events are ignored for the moment
+		s.Log.Debug("Received keepalive event")
+
+	default:
+		// Received a yet unsupported event type
+		s.Log.Debug("Received unsupported event")
+	}
+}
+
+// Init is for setup, and validating config.
+func (s *ctrlXDataLayerSSE) Init() error {
+	if s.Timeout == 0 {
+		s.Timeout = defaultTimeout
+	}
+	if time.Duration(s.Timeout) < time.Duration(defaultTimeout) {
+		s.Log.Errorf("Setting timeout must be at least %v seconds", time.Duration(defaultTimeout).Seconds())
+		return errors.New("timeout configuration error")
+	}
+
+	for i, sub := range s.Subscription {
+		err := choice.Check(sub.QueueBehaviour, queueBehaviours)
+		if err != nil {
+			s.Log.Errorf("SSE subscription %v: setting 'queue_behaviour' %v is incorrectly configured", i, sub.QueueBehaviour)
+			s.Log.Infof("The right queue behaviour values are %v", queueBehaviours)
+			return err
+		}
+		err = choice.Check(sub.ValueChange, valueChanges)
+		if err != nil {
+			s.Log.Errorf("SSE subscription %v: setting 'value_change' %v is incorrectly configured", i, sub.ValueChange)
+			s.Log.Infof("The right value change values are %v", valueChanges)
+			return err
+		}
+	}
```

Comment:
Please reduce the amount of logging!
```suggestion
	for i, sub := range s.Subscription {
		if err := choice.Check(sub.QueueBehaviour, queueBehaviours); err != nil {
			return fmt.Errorf("subscription %v: 'queue_behaviour' %v invalid", i, sub.QueueBehaviour)
		}
		if err := choice.Check(sub.ValueChange, valueChanges); err != nil {
			return fmt.Errorf("subscription %v: 'value_change' %v invalid", i, sub.ValueChange)
		}
	}
```